### PR TITLE
cleanup(metrics): add original rule name as label

### PR DESCRIPTION
**Pull request by incertum**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

> /kind release

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

/area engine

> /area tests

> /area proposals

> /area CI

**What this PR does / why we need it**:

cleanup(metrics): add original rule name as label

https://github.com/falcosecurity/charts/issues/672#issuecomment-2118685055 @leogr 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
cleanup(metrics): add original rule name as label
```

### CODE REVIEW

File: userspace/falco/falco_metrics.cpp [hunk 1]
```diff
@@ -182,6 +182,7 @@ std::string falco_metrics::to_text(const falco::app::state& state)
 																	rules_by_id[i]->load());
 				prometheus_metrics_converter.convert_metric_to_unit_convention(metric);
 				const std::map<std::string, std::string>& const_labels = {
+					{"rule", rule->name},
```
Comment (incertum) [inline userspace/falco/falco_metrics.cpp:185]: Useful to preserve the original un-sanitized rule name for prometheus.

File: userspace/falco/stats_writer.cpp [hunk 2]
```diff
@@ -340,7 +340,7 @@ void stats_writer::collector::get_metrics_output_fields_wrapper(
 	{
 		fs::path fs_path = item.first;
 		std::string metric_name_file_sha256 = fs_path.filename().stem();
-		metric_name_file_sha256 = "falco.sha256_rule_file." + falco::utils::sanitize_metric_name(metric_name_file_sha256);
+		metric_name_file_sha256 = "falco.sha256_rules_file." + falco::utils::sanitize_metric_name(metric_name_file_sha256);
```
Comment (incertum) [inline userspace/falco/stats_writer.cpp:343]: We changed it to plural form elsewhere let's be consistent.

**Review by FedeDP — APPROVED** [review]
/approve

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**